.github/workflows/_ci.yaml

```diff
@@ -21,20 +21,107 @@ on:
+      TEST_TRITON: ${{ steps.run-conditions.outputs.TEST_TRITON }}
+      TEST_MAXTEXT: ${{ steps.run-conditions.outputs.TEST_MAXTEXT }}
+      TEST_LEVANTER: ${{ steps.run-conditions.outputs.TEST_LEVANTER }}
+      TEST_UPSTREAM_PAX: ${{ steps.run-conditions.outputs.TEST_UPSTREAM_PAX }}
+      TEST_UPSTREAM_T5X: ${{ steps.run-conditions.outputs.TEST_UPSTREAM_T5X }}
+      TEST_ROSETTA_PAX: ${{ steps.run-conditions.outputs.TEST_ROSETTA_PAX }}
+      TEST_ROSETTA_T5X: ${{ steps.run-conditions.outputs.TEST_ROSETTA_T5X }}
+    steps:
+      - name: Check out the repository under ${GITHUB_WORKSPACE}
+        uses: actions/checkout@v4
+
+      - name: Validate input `TEST_SUBSET`
+        shell: bash
+        id: validate
+        run: |
+          test_subsets=$(echo "${{ inputs.TEST_SUBSET }}" | awk '{$1=$1;print}')
+          IFS=' ' read -a test_subsets <<< "$test_subsets"
+
+          valid_test_subsets=$(yq '.on.workflow_dispatch.inputs.TEST_SUBSET.options | join(",")'  .github/workflows/ci.yaml)
+          
+          for test_subset in "${test_subsets[@]}"; do     
+            if [[ $test_subset == "" || "$valid_test_subsets" != *"$test_subset"* ]]; then
+              message="Invalid value for '<TEST_SUBSET>' provided ❌. Expected one of: ($valid_test_subsets), Actual: '$test_subset'"
+              exit 1
+            fi
+          done
+
+          test_subsets=$(IFS=' '; echo "${test_subsets[*]}")
+          echo "test_subset=$test_subsets" >> "$GITHUB_OUTPUT"
+
+      - name: Compile run-conditions
+        id: run-conditions
+        shell: bash
+        run: |
+          if_testset_in() {
+            local list_of_items="$1"  # The list of items as a string
+            IFS=' ' read -a test_subsets <<< "${{ steps.validate.outputs.test_subset }}"
 
+            contains=0
+            for test_subset in "${test_subsets[@]}"; do     
+              if [[ $test_subset == "" || "$list_of_items" != *"$test_subset"* ]]; then
+                contains=$(( $contains | 0 ))
+              else
+                contains=$(( $contains | 1 ))
+              fi
+            done
+
+            [[ $contains -eq 1 ]] && echo "true" || echo "false"
+          }
+          
+          echo BUILD_TRITON=$(if_testset_in "base jax triton") >> $GITHUB_OUTPUT
+          echo BUILD_EQUINOX=$(if_testset_in "base jax equinox") >> $GITHUB_OUTPUT
+          echo BUILD_MAXTEXT=$(if_testset_in "base jax maxtext") >> $GITHUB_OUTPUT
+          echo BUILD_LEVANTER=$(if_testset_in "base jax levanter") >> $GITHUB_OUTPUT
+          echo BUILD_T5X=$(if_testset_in "base jax upstream-t5x rosetta-t5x") >> $GITHUB_OUTPUT
+          echo BUILD_PAX=$(if_testset_in "base jax upstream-pax rosetta-pax") >> $GITHUB_OUTPUT
+          echo BUILD_GROK=$(if_testset_in "base jax grok") >> $GITHUB_OUTPUT
+
+          echo TEST_JAX=$(if_testset_in "base jax") >> $GITHUB_OUTPUT
+          echo TEST_TRITON=$(if_testset_in "base jax triton") >> $GITHUB_OUTPUT
```

Comment:
Looks like you don't need to do the comparison again: if you build something, you always want to test it. IMHO, you can do this:
```
echo "TEST_TRITON=${BUILD_TRITON}" >> $GITHUB_OUTPUT
```
Same for other tests.

Reply:
You are right, some of these checks we can simplify. However, if differs for rosetta-tests. Here, we need to build upstream but want to test rosetta only. But I will still simplify it where possible